crypto/keys/keys_test.go

```diff
@@ -141,57 +142,14 @@ func TestLoadPrivateKeyAndSign(t *testing.T) {
 			continue
 		}
 
-		signature, err := k.Sign(rand.Reader, digest, hasher)
-		if err != nil {
-			t.Errorf("%v: failed to sign: %v", test.name, err)
-			continue
-		}
-
-		// Do a round trip by verifying the signature using the public key.
-		if err := verify(k.Public(), digest, signature, hasher, hasher); err != nil {
+		// Check the key by creating a signature and verifying it.
+		if err := signAndVerify(k, k.Public()); err != nil {
 			t.Errorf("%v: %v", test.name, err)
 		}
-
-	}
-}
-
-func TestSignatureAlgorithm(t *testing.T) {
-	tests := []struct {
-		name   string
-		keyPEM string
-		want   sigpb.DigitallySigned_SignatureAlgorithm
-	}{
-		{
-			name:   "ECDSA",
-			keyPEM: ecdsaPublicKey,
-			want:   sigpb.DigitallySigned_ECDSA,
-		},
-		{
-			name:   "RSA",
-			keyPEM: rsaPublicKey,
-			want:   sigpb.DigitallySigned_RSA,
-		},
-		{
-			name:   "DSA",
-			keyPEM: dsaPublicKey,
-			want:   sigpb.DigitallySigned_ANONYMOUS,
-		},
-	}
-
-	for _, test := range tests {
-		key, err := NewFromPublicPEM(test.keyPEM)
-		if err != nil {
-			t.Errorf("%v: Failed to load key: %v", test.name, err)
-			continue
-		}
-
-		if got := SignatureAlgorithm(key); got != test.want {
-			t.Errorf("%v: SignatureAlgorithm(%v) = %v, want %v", test.name, key, got, test.want)
-		}
 	}
 }
 
-func TestGenerateKey(t *testing.T) {
+func TestNewFromSpec(t *testing.T) {
```

Comment:
Done for TestNewFromSpec. I was intending to do this more thoroughly in the follow-up CL by replacing TestLoadPrivateKeyAndSign with individual tests for each of the public funcs (and put those tests in the appropriate _test.go file).